src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonProcessor.java

```diff
@@ -108,13 +110,13 @@ public Content generatePackagesFromComponents(final Repository repository, final
       throws IOException
   {
     return buildPackagesJson(repository, StreamSupport.stream(components.spliterator(), false)
-        .map(component -> component.group() + "/" + component.name()).collect(Collectors.toList()));
+        .map(component -> component.group() + "/" + component.name()).collect(Collectors.toSet()));
```

Comment:
This is just me saying I know very little about Composer, it doesn't require the version in the packages.json?

Reply:
> This is just me saying I know very little about Composer, it doesn't require the version in the packages.json?

No, not at this level. This level basically points to another file that gives you the versions for a particular package.

Reply:
The wild wacky world of formats!